bson/unmarshal.go

```diff
@@ -58,8 +58,8 @@ func UnmarshalValue(t Type, data []byte, val interface{}) error {
 // UnmarshalExtJSON parses the extended JSON-encoded data and stores the result
 // in the value pointed to by val. If val is nil or not a pointer, UnmarshalExtJSON
 // returns an error.
```

Comment:
```suggestion
// returns an error.
//
// If canonicalOnly is true, UnmarshalExtJSON returns an error if the Extended
// JSON was not marshaled in canonical mode.
//
// For more information about Extended JSON, see
// https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/
```

I recommend extending the documentation here to mention the behavior of `canonicalOnly`.